### ISSUE TITLE
Failed notification shown for AirPods

### ISSUE THREAD

**Issue by tbrek** [tilmanginzel/alfred-bluetooth-workflow#10]
When trying to connect my AirPods I'm getting an error message, but they connect successfully:

<img width="371" alt="image" src="https://user-images.githubusercontent.com/9820995/66211638-5b31c180-e6b4-11e9-96fd-556cce112c7d.png">

<img width="587" alt="image" src="https://user-images.githubusercontent.com/9820995/66211655-6389fc80-e6b4-11e9-84c1-cfd554a2ba8b.png">

**Comment by tilmanginzel** [comment]
Hi @tbrek, thanks for the issue. This is a bit weird.

The code where that notification is generated is in [`blt.sh`](https://github.com/tilmanginzel/alfred-bluetooth-workflow/blob/master/scripts/blt.sh):
```bash
    ../blueutil --connect ${DEVICE}

    if [[ $(../blueutil --is-connected ${DEVICE}) -eq 1 ]]
    then
        ./notify.sh -m "Connected to ${TITLE}" -i success
    else
        ./notify.sh -m "Failed to connect to ${TITLE}. Click to retry." -i failure -c "blt.sh ${DEVICE} \"${TITLE}\""
    fi
```

So blueutil apparently says that the device is not connected. Can you please check the following:

* Locate the workflow directory
* Open a terminal
* Find the device id, e.g. via `./blueutil --recent`
* Check the output of `./blueutil --is-connected <device_id>`

Is the output `1` or `0`?

**Comment by tbrek** [comment]
```
@./blueutil --recent
address: 88-64-40-e1-f8-5e, connected (master, -52 dBm), not favourite, paired, name: "Tom’s AirPods", recent access date: 2019-10-07 09:48:19 +0000

$ ./blueutil --is-connected 88-64-40-e1-f8-5e
1
```

It looks like blueutil sees it as connected.

**Comment by tbrek** [comment]
I think I fixed it. Adding `sleep 1` helps as the script seems to be too fast, so, if `blt.sh`
is:
```
#!/usr/bin/env bash

PARENT_PATH=$(cd "$(dirname "${BASH_SOURCE[0]}")"; pwd -P)
cd "${PARENT_PATH}"

DEVICE=$1
TITLE=$2

if [[ $(../blueutil --is-connected ${DEVICE}) -eq 1 ]]
then
    ../blueutil --disconnect ${DEVICE}

    if [[ $(../blueutil --is-connected ${DEVICE}) -eq 0 ]]
    then
        ./notify.sh -m "Disconnected from ${TITLE}"
    else
        ./notify.sh -m "Failed to disconnect ${TITLE}. Click to retry." -i failure -c "blt.sh ${DEVICE} \"${TITLE}\""
    fi
else
    ../blueutil --connect ${DEVICE}
    sleep 1
    if [[ $(../blueutil --is-connected ${DEVICE}) -eq 1 ]]
    then
        ./notify.sh -m "Connected to ${TITLE}" -i success
    else
        ./notify.sh -m "Failed to connect to ${TITLE}. Click to retry." -i failure -c "blt.sh ${DEVICE} \"${TITLE}\""
    fi
fi
```
It works fine for me.

**Comment by tilmanginzel** [comment]
@tbrek Hm that is interesting. If I am understanding the code behind blueutil correctly, it calls this API: [IOBluetoothDevice.openConnection()](https://developer.apple.com/documentation/iobluetooth/iobluetoothdevice/1428435-openconnection).

Quote from the documentation:

> This method is synchronous and will not return until either a connection has been established or the create connection has failed (perhaps timed out).

Afterwards, [IOBluetoothDevice.isConnected()](https://developer.apple.com/documentation/iobluetooth/iobluetoothdevice/1432785-isconnected) is called, which theoretically _should_ yield the same result.

Maybe I should try to not verify the connection via `--is-connected`, but rather check the exit code of the `--connect` command. It returns either 0 or 1, so should be easy to check. This would be a) faster, and b) hopefully more stable.

**Comment by tilmanginzel** [comment]
@tbrek Can you check if the following version does the trick (without any sleep)? :)

[Bluetooth Connector.alfredworkflow.zip](https://github.com/tilmanginzel/alfred-bluetooth-workflow/files/3698562/Bluetooth.Connector.alfredworkflow.zip)

**Comment by avxkim** [comment]
@tilmanginzel your archive complains regards etree.so and doesn't open.

**Comment by tbrek** [comment]
> @tbrek Can you check if the following version does the trick (without any sleep)? :)
> 
> [Bluetooth Connector.alfredworkflow.zip](https://github.com/tilmanginzel/alfred-bluetooth-workflow/files/3698562/Bluetooth.Connector.alfredworkflow.zip)

Yes. That also fixed the issue. Thanks.

**Comment by avxkim** [comment]
@tbrek don't you have issues with etree.so with the above workflow?

**Comment by tbrek** [comment]
@webcoderkz no i don't. Could be Catalina related. I have not upgraded yet.

**Comment by avxkim** [comment]
Ah, i see, i thought you're on Catalina already.

@tilmanginzel please, update your latest workflow for Catalina. Because older version of workflow works (i just updated it using `workflow:update` to the latest). If i delete the old workflow, and download 0.5.0 it gives me etree.so error.

**Comment by tilmanginzel** [comment]
> Yes. That also fixed the issue. Thanks.

Perfect! I have pushed the fix and released version 0.6.0 :)

@webcoderkz will comment in #9.
